stock_reorder_forecast/models/product_product.py

```diff
@@ -0,0 +1,188 @@
+# -*- coding: utf-8 -*-
+# © 2012-2016 Therp BV <http://therp.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from datetime import date, timedelta
+from openerp import api, fields, models
+from openerp.tools import float_round
+from openerp.addons.decimal_precision import get_precision
+
+
+class ProductProduct(models.Model):
+
+    _inherit = 'product.product'
+
+    # Defauts if not specified
+
+    TURNOVER_PERIOD = 365
+    STOCK_PERIOD_MIN = 91
+    STOCK_PERIOD_MAX = 185
+    PURCHASE_MULTIPLE = 1.0
+
+    stock_period_max = fields.Integer(
+        'Maximium stock', help='Maximum stock in days turnover to resupply '
+        'for. Used by the purchase proposal.')
+    turnover_period = fields.Integer(
+        string='Turnover period',
+        default=90,
+        help='Turnover days to calculate average '
+        'turnover per day. Used by the purchase proposal.')
+    turnover_average = fields.Float(
+        'Turnover per day', digits=get_precision('Purchase Price'),
+        readonly=True, help='Average turnover per day. Used by the purchase '
+        'proposal.')
+    ultimate_purchase = fields.Date(
+        'Ultimate purchase', readonly=True, help='Ultimate date to purchase '
+        'for not running out of stock. Used by the purchase proposal.')
+    automatic_purchase_days = fields.Integer(
+        "Trigger automatic purchase proposal on first supplier",
+        default=0,
+        help="Trigger an automatic RFQ on first (default supplier) if ultimate"
+             "purchase is less than these days away"
+    )
+
+    def isbuy(self, product):
```

Comment:
ormcache this function. Or make this a stored field on products recomputed when the routes change